src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -1864,24 +1864,12 @@ bool G1CollectedHeap::try_collect_fullgc(GCCause::Cause cause,
                                          const G1GCCounters& counters_before) {
   assert_heap_not_locked();
 
-  while(true) {
-    VM_G1CollectFull op(counters_before.total_collections(),
-                        counters_before.total_full_collections(),
-                        cause);
-    VMThread::execute(&op);
-
-    // Request is trivially finished.
-    if (!GCCause::is_explicit_full_gc(cause) || op.gc_succeeded()) {
-      return op.gc_succeeded();
```

Comment:
Complete removal of this part isn't correct.  The premise of this change, that "full-gc always
run-to-completion" is not correct.  A `_g1_periodic_collection` may be cancelled:
https://github.com/openjdk/jdk/blob/50a7c61d28b9885ff48f4fcd8bfd460b507bbcef/src/hotspot/share/gc/g1/g1VMOperations.cpp#L39-L48
Such a collection is not an explicit full-gc, so that part of the test may be true, when this function
returns false because the gc did not succeed.

Reply:
Updated the description. The overall result of `!GCCause::is_explicit_full_gc(cause) || op.gc_succeeded()` is always true.

Reply:
The pre-existing code seems to be buggy: if the operation had been skipped, it should have returned false given the one caller that reads the boolean result (periodic collection invocation) to print a log line. I.e. maybe something like:

`return op.prologue_succeeded() && op.gc_succeeded();`

Similar to `G1CollectedHeap::do_collection_pause()`.

Reply:
(Somehow comments seem to have disappeared, probably my fault. anyway:)

This method returns `false` if the VM op has not been executed (skipped). The new code always returns `true`, which is wrong afaict.